metric_providers/powermetrics/provider.py

```diff
@@ -55,17 +57,21 @@ def stop_profiling(self):
 
         # As killall returns right after sending the SIGKILL we need to wait and make sure that the process
         # had time to flush everything to disk
-        count = 0
-        while self.is_powermetrics_running():
-            print(f"Waiting for powermetrics to shut down (try {count}/60). Please do not abort ...")
-            time.sleep(1)
-            count += 1
-            if count >= 60:
-                subprocess.check_output('sudo /usr/bin/killall -9 powermetrics', shell=True)
-                raise RuntimeError('powermetrics had to be killed with kill -9. Values can not be trusted!')
-
-        # We need to give the OS a second to flush
-        time.sleep(1)
+        if self._skip_check:
```

Comment:
If you run into a Permission error in the block before this (the except block) you would still kill the process, which is likely not what you want with that code. I think that should also be guard-claused.